pkg/net/url.go

```diff
@@ -100,6 +101,11 @@ func ParseURL(s string) (u *URL, err error) {
 		uu.Path = path.Clean(uu.Path)
 	}
 
+	// path.Clean removes the trailing '/' and converts '//' to '/'.
+	if strings.HasSuffix(s, "/") && !strings.HasSuffix(uu.Path, "/") {
```

Comment:
@Praveenrajmani I doubt cleaning uu.Path has any useful purpose, I think it is enough to just remove it

Reply:
@vadmeste In windows, filepath,Clean() converts all the `/` into `\`. And it resolves all the `.` and `..` to the current and parent directories as well.

Reply:
And i fear it would break in many places. It might be a breaking change

Reply:
This `URL` package is little different than standard `url.URL` i.e. it removes some corner limitation of `url.URL`. This `URL` package ensures below
1. Not allowed to have scheme without host.
2. port in host should be a number.
3. Keeps path cleaned e.g. `/mybucket//myobject/` is converted to `/mybucket/myobject/` (extra slash are moved)
4. Removes default ports 80/443 for schemes `http/https` when converting to string.

When these are not benefited, its better to use `url.URL` directly in those cases.